### ISSUE TITLE
PHPCS task does not output errors

### ISSUE THREAD

**Issue by ctrl-f5** [phpro/grumphp#523]
| Q               | A
| --------------- | ---
| Version         | 0.14.1
| Question?       | yes
| Related issues | #487, #275

when running the phpcs task, I'm simply getting an X, but no actual errors are shown, is this intended behavior?
```
$ include/vendor/bin/grumphp run
GrumPHP is sniffing your code!
Running task 1/1: Phpcs... ✘

                                    _
                                   | |
 _   _  ___  _   _   ___ _   _  ___| | __
| | | |/ _ \| | | | / __| | | |/ __| |/ /
| |_| | (_) | |_| | \__ \ |_| | (__|   <
 \__, |\___/ \__,_| |___/\__,_|\___|_|\_\
  __/ |
 |___/
```

**Comment by veewee** [comment]
Hi @ctrl-f5,

It should output the error message. Can you run the same command by adding the options `-vvv` to the cli. That way you can debug what command is executed for phpcs and why it does not output error messages.

**Comment by veewee** [comment]
Hi @ctrl-f5,

Internally, it uses a formatter to parse the json to a text format:
https://github.com/phpro/grumphp/blob/master/src/Formatter/PhpcsFormatter.php

The format might have changed in a newer version of phpcs. Can you share me the version of phpcs?

**Comment by ctrl-f5** [comment]
```
$ composer info squizlabs/php_codesniffer
name     : squizlabs/php_codesniffer
descrip. : PHP_CodeSniffer tokenizes PHP, JavaScript and CSS files and detects violations of a defined set of coding standards.
keywords : phpcs, standards
versions : * 3.2.3

```

**Comment by ctrl-f5** [comment]
![ouput](https://user-images.githubusercontent.com/485346/42499574-6c6d3152-842f-11e8-9c6f-4ee0eeb5de0a.png)

It isn't getting any output from the process, I'll debug it further

**Comment by ctrl-f5** [comment]
seems to be due to the sheer number of files or size of the output, when I restrict phpcs to just the config folder, it works as expected:

```
    phpcs:
      standard: "./dev/phpcs/phpcs.xml"
      whitelist_patterns:
        - /^config\/(.*)/
```

When doing a run of the complete codebase, it's appending more than 10K files to the command (command length is 650K chars)

**Comment by veewee** [comment]
Hi @ctrl-f5,

It seems like you are hitting the maximum CLI input length. 
We could avoid adding all files to the cli command during run context, but then the whitelist logic would be broken.

**Comment by arsonik** [comment]
any update on that issue ?

**Comment by ryantfowler** [comment]
@ctrl-f5 I've recently ran into a similar issue, but in my case, I tested the following:
1. edit a `.php` file, which is configured in my `grumphp.yml` file as whitelisted ( I don't have any reason to believe that the whitelisting actually matters, but adding that detail here for thoroughness ), and consciously add code formatting which violates the phpcs ruleset configured in the phpcs task in my `grumphp.yml` file.
2. Execute `git commit -am "JIRA-123: short description"`
3. The output I receive is that the phpcs task completed successfully, no errors found.
4. Rollback the file ( `git reset --soft HEAD~1 && git reset HEAD <path-to-file>` )
5. Execute `git add <path-to-file> && git commit -m "JIRA-123: short description"`
6. The output I receive is as expected, and `phpcs` reports the appropriate violations of the configured phpcs ruleset.

It appears as though the problem surrounds the use of `git commit -am` vs `git add` followed by `git commit`. I'm working with a Magento 2 codebase, so it's a large volume of files. To further validate this, when I execute `./vendor/bin/grumphp run` I receive the expected output, so I'm not impeded due to a large volume of files.

@ctrl-f5 would you be able to possibly try running through your test case again using `git add && git commit` vs `git commit -am`? Or if you are already using `git add && git commit` vs `git commit -am`, could you add that detail here?

Thank you
Cc: @veewee

**Comment by veewee** [comment]
Fixed in #796
